manual/trees/index.md

```diff
@@ -517,6 +511,44 @@ The scatterplot is drawn.
 Note that not each `(x,y) point on a scatterplot represents two values in your N−tuple. In fact, the scatterplot is a grid and each square in
 the grid is randomly populated with a density of dots that’s proportional to the number of values in that grid.
 
+### Indexing trees
+
+- Use [TTree::BuildIndex()](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bd0b06eea6c12b998){:target="_blank"} method to build an index table using expressions depending on the value in the leaves.
+
+The index is built in the following way:
+- A pass on all entries is made like in [TTree::Draw()](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bTTree::Draw()d0b06eea6c12b998){:target="_blank"}.
+- `var1` = `majorname`
+- `var2` = `minorname`
+- `sel = 231` × _majorname_ + _minorname_
+- For each entry in the tree the `sel` expression is evaluated and the result array is sorted into `fIndexValues`.
+
+Once the index is calculated, an entry can be retrieved with
+[TTree::GetEntryWithIndex(majornumber, minornumber)](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bTTree::Draw()d0b06eea6c12b998){:target="_blank"}.
+
+_**Example**_
+
+{% highlight C++ %}
+// To create an index using the leaves "Run" and "Event".
+   tree.BuildIndex("Run","Event");
+
+// To read entry corresponding to Run=1234 and Event=56789.
+   tree.GetEntryWithIndex(1234,56789);
+   {% endhighlight %}
+
+Note that `majorname` and `minorname` can be expressions using original tree variables e.g., `"run-90000"` or `"event +3*xx"`.
+
+In case an expression is specified, the equivalent expression must be computed when calling
+[TTree::GetEntryWithIndex(majornumber, minornumber)](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bTTree::Draw()d0b06eea6c12b998){:target="_blank"}.
+To build an index with only `majorname`, specify `minorname="0"` (default).
+
+Once the index is built, it can be saved with the `TTree` object with `tree.Write()`.
+
+The most convenient place to create the index is at the end of the filling process just before saving the tree header. If a previous index was calculated, it will be redefined by this new call.
```

Comment:
```suggestion
The most convenient place to create the index is at the end of the filling process just before saving the tree header. If a previous index was available, it will be overridden by this new call.
```